cider/type/plan/UnaryExpr.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include "UnaryExpr.h"
+#include "exec/template/Execute.h"
+
+namespace Analyzer {
+using namespace cider::jitlib;
+
+JITExprValue& UOper::codegen(JITFunction& func) {
+  if (auto expr_var = get_expr_value()) {
+    return *expr_var;
+  }
+  auto operand = const_cast<Analyzer::Expr*>(get_operand());
+  if (is_unnest(operand) || is_unnest(operand)) {
+    CIDER_THROW(CiderCompileException, "Unnest not supported in comparisons");
+  }
+  const auto& operand_ti = operand->get_type_info();
+  if (operand_ti.is_decimal() || operand_ti.is_timeinterval()) {
+    CIDER_THROW(CiderCompileException,
+                "Decimal and TimeInterval are not supported in arithmetic codegen now.");
+  }
+  auto& operand_val = operand->codegen(func);
+  switch (get_optype()) {
+    case kISNULL: {
+      return set_expr_value(JITValuePointer(operand_val.getNull()));
```

Comment:
need to also `set_nulls()` here?

Reply:
There is no logic for null set now and @bigPYJ1151 will cover this in his next PR. Will add this part later.